libs/payments/ui/src/lib/server/purchase-details.tsx

```diff
@@ -5,11 +5,57 @@ import {
   getLocalizedCurrency,
   getLocalizedCurrencyString,
 } from '@fxa/shared/l10n/fluent';
+import { FluentBundle } from '@fluent/bundle';
 import { headers } from 'next/headers';
 import Image from 'next/image';
 import { formatPlanPricing } from '../utils/helpers';
+import '../../styles/index.css';
 
-export type PurchaseDetailsProps = {
+type ListLabelItemProps = {
```

Comment:
Added `<ListLabelItem>` to handle how we display the list of amounts that make up the total amount. This came via a suggestion from @xlisachan ([Original code here](https://github.com/mozilla/fxa/blob/e03889f7edb533ec7d9b014875f7afc73d499c10/packages/gatsby-prototype/src/lib/utils.tsx))

I opted to keep it inside `purchase-details.tsx` since it's currently only used in `<PurchaseDetails>`. If `<ListLabelItem>` becomes reusable in other components as well, I was thinking we could move it into it's own file at that time. (For clarity's sake though, I'm also open to moving it into it's own file now. Lets discuss!)

Reply:
Thinking that moving into its own file would be better. Unless we plan to do something different for Subscription Management when we get to it.

In addition, consider in future we might opt to display summary or invoice information and moving it now would alleviate the extra work needed to move it later, as well as prevent duplicate information from being produced (I find a lot of that has happened with our current structure, and with the amount of effort involved to clean it up, it's all been backlogged).

Reply:
Tis true. I find that our instincts are to create things before checking if it already exists elsewhere, which has led to a lot of duplication/redundancies. It might be worth establishing the location of all the common assets, helpers, etc, now to break that habit early on in SP3.

Reply:
With the SP3 project there are a couple of considerations and goals that we had in mind.
* Reduce the number of components
* Don't optimize too early (i.e. do it when it's necessary)

I think this component is a good example of that. Currently the ListLabelItem is only used in Plan Details. I don't actually see it being used anywhere else. I don't think we have another place in SubPlat where we have a label and then price. (Except in emails)

But if that changes, and we do need it somewhere else, then I think we should pull it into a separate file.

All that said, @xlisachan as you said, we should find locations for common assets, etc. and would be good to discuss that. Maybe we can have a proposal and discuss it at our next Engineering Weekly?

Reply:
I am aligned with keeping `ListLabelItem` within `PurchaseDetails` for now while keeping in mind that there is a potential to re-use the `ListLabelItem` helper in the Upgrade page, where we similarly list price items as well.

If we find that we can re-use this helper component, I'd prefer it placed within a helper file (perhaps in `payments-ui`) among other helper functions that are reused throughout payments, rather than it be its own component file. There is also flexibility to revisit having a helper file vs individual components files at a later time if the file gets lengthy.

Happy to discuss a proposal for a location for common assets, etc during our next Engineering Weekly!